src/nni_manager/training_service/remote_machine/remoteMachineTrainingService.ts

```diff
@@ -327,19 +350,59 @@ class RemoteMachineTrainingService implements TrainingService {
         return deferred.promise;
     }
 
-    public cleanUp(): Promise<void> {
+    public async cleanUp(): Promise<void> {
         this.log.info('Stopping remote machine training service...');
         this.stopping = true;
+        let timeoutId: NodeJS.Timer;
+        const delay: Promise<{}> = new Promise((resolve: Function, reject: Function): void => {
+            timeoutId = setTimeout(
+                () => { reject(new Error('TrainingService cleanup timeout.')); },
+                10000);
```

Comment:
There is a delay function in utils.ts,  no need to create a new one.

Reply:
updated to use utils.delay()